azext_iot/tests/test_iot_ext_int.py

```diff
@@ -71,6 +71,33 @@ def test_hub(self):
             expect_failure=True,
         )
 
+        # Test 'az iot hub connection-string show'
+        conn_str_pattern = r'^HostName={0}.azure-devices.net;SharedAccessKeyName=iothubowner;SharedAccessKey='.format(
+            LIVE_HUB)
+        conn_str_eventhub_pattern = r'^Endpoint=sb://'
+        self.cmd('iot hub connection-string show -n {0}'.format(LIVE_HUB), checks=[
+            self.check_pattern('connectionString', conn_str_pattern)
+        ])
+
+        self.cmd('iot hub connection-string show -n {0} --eh'.format(LIVE_HUB), checks=[
+            self.check_pattern('connectionString', conn_str_eventhub_pattern)
+        ])
+
+        self.cmd('iot hub connection-string show -n {0} -g {1}'.format(LIVE_HUB, LIVE_RG), checks=[
+            self.check('length(@)', 1),
+            self.check_pattern('connectionString', conn_str_pattern)
+        ])
+
+        self.cmd('iot hub connection-string show -n {0} -g {1} --all'.format(LIVE_HUB, LIVE_RG), checks=[
+            self.check('length(connectionString[*])', 5),
```

Comment:
I'm uneasy asserting a particular length for the connectionString list, because for example if your test hub has an extra policy (for whatever reason) this test will fail. I think its safer to check if the length of the list is >= 1, because at least one policy should exist to interact with the iothub dataplane.

Reply:
Done